lib/support_rota_to_productive/support_rotation.rb

```diff
@@ -15,25 +15,37 @@ def hash
 
     class << self
       def from_support_rota
-        result = []
-        result << JSON.parse(client.get(developer_endpoint)) if ENV["IMPORT_DEV_IN_HOURS"].eql?("true")
-        result << JSON.parse(client.get(ops_endpoint)) if ENV["IMPORT_OPS_IN_HOURS"].eql?("true")
+        @from_support_rota ||= begin
+          result = []
+          result << JSON.parse(client.get(developer_endpoint)) if ENV["IMPORT_DEV_IN_HOURS"].eql?("true")
+          result << JSON.parse(client.get(ops_endpoint)) if ENV["IMPORT_OPS_IN_HOURS"].eql?("true")
 
-        result.flatten.map do |support_day|
-          new(values_for(support_day))
+          result.flatten.map do |support_day|
+            new(values_for(support_day))
+          end
         end
       end
 
       def from_productive
-        Productive::Booking.where(project_id: SUPPORT_PROJECT_ID, after: Date.today).all.map(&:to_support_rotation)
+        bookings = Productive::Booking
+          .where(project_id: SUPPORT_PROJECT_ID, after: date_of_earliest_support_event)
+          .all
+        bookings.map(&:to_support_rotation)
+      end
+
+      def date_of_earliest_support_event
+        first_support_date = from_support_rota.first
+        return Date.today unless first_support_date.respond_to?(:date)
```

Comment:
Is the only circumstance that the object doesn't respond to `#date` when it's `nil`? I can't work out any other path to it.

If it's only for nil checking, would it be clearer to use something like `if from_support_rota.empty?`

Reply:
In the end this method has been removed so I took no further action.